src/components/QuestionnaireForm/QuestionnaireForm.jsx

```diff
@@ -698,9 +698,45 @@ export default class QuestionnaireForm extends Component {
     return c;
   }
 
+  addAuthorToResponse(qr, practitionerRef) {
+    function traverseToItemsLeafNode(item, practitionerRef) {
+      if (!item.item) {
+        return addAuthor(item, practitionerRef);
+      }
+      else {
+        item.item.map(item => {
+          traverseToItemsLeafNode(item, practitionerRef);
+        })
+      }
+    }
+    // url is a string
+    function addAuthor(item, practitionerRef) {
+      var url = "http://hl7.org/fhir/StructureDefinition/questionnaireresponse-author"
+      const urlValRef =
+      {
+        "url": url,
+        "valueReference":
+        {
+          "reference": practitionerRef
+        }
+      }
+      if (item.extension) {
+        item.extension.push(urlValRef)
```

Comment:
Do we need to check if the same extension already exists? if it is already there, we should not push it again.

Reply:
Hi Yolanda, this is checking to see if there is already an extension. If there is, then we add the urlValRef to it, and if not, we create one.

Reply:
The if logic is fine. I meant to check if item.extension already has "questionnaireresponse-author" extension. If it already has one, push a new one will either create a duplicate extension with the same value or an extension with the same url and different reference.

Reply:
ah got it! just added that in!